File: frontend/smart-library-frontend/src/services/bookService.js

import api from './api';

export const getAllBooks = (page = 0, size = 10, sortBy = 'createdAt', direction = 'desc') =>
  api.get(`/books?page=${page}&size=${size}&sortBy=${sortBy}&direction=${direction}`);
export const searchBooks = (keyword, page = 0, size = 10) =>
  api.get(`/books/search?keyword=${encodeURIComponent(keyword)}&page=${page}&size=${size}`);
export const getBookById = (id) => api.get(`/books/${id}`);
export const getBookByIsbn = (isbn) => api.get(`/books/isbn/${isbn}`);
export const getAllCategories = () => api.get('/books/categories');
export const getBooksByCategory = (cat, page = 0, size = 10) =>
  api.get(`/books/category/${encodeURIComponent(cat)}?page=${page}&size=${size}`);
export const getRecentBooks = (limit = 10) => api.get(`/books/recent?limit=${limit}`);
export const addBook = (data) => api.post('/books', data);
export const updateBook = (id, data) => api.put(`/books/${id}`, data);
export const deleteBook = (id) => api.delete(`/books/${id}`);

File: backend/smart-library-backend/src/main/java/com/library/service/BookService.java
package com.library.service;

import com.library.dto.request.BookRequest;
import com.library.dto.response.BookResponse;
import com.library.dto.response.PageResponse;
import org.springframework.data.domain.Pageable;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface BookService {
    BookResponse addBook(BookRequest request);
    BookResponse updateBook(Long id, BookRequest request);
    void deleteBook(Long id);
    BookResponse getBookById(Long id);
    BookResponse getBookByIsbn(String isbn);
    PageResponse<BookResponse> getAllBooks(Pageable pageable);
    PageResponse<BookResponse> searchBooks(String keyword, Pageable pageable);
    PageResponse<BookResponse> getBooksByCategory(String category, Pageable pageable);
    List<String> getAllCategories();
    List<BookResponse> getRecentBooks(int limit);
}
